### PR TITLE
expose profiler to LM hooks

### PR DESCRIPTION
**Pull request by andre-merzky**

### CODE REVIEW

File: src/radical/pilot/agent/lm/orte.py [hunk 1]
```diff
@@ -180,9 +184,12 @@ def lrms_shutdown_hook(cls, name, cfg, lrms, lm_info, logger):
...
                 if not orterun:
                     raise Exception("Couldn't find orterun")
-                subprocess.Popen([orterun, "--hnp", lm_info['dvm_uri'], "--terminate"])
+                ru.sh_callout('%s --hnp %s --terminate' 
+                             % (orterun, lm_info['dvm_uri']))
+                profiler.prof(event='orte_dvm_stop', uid=cfg['pilot_id'])
             except Exception as e:
-                logger.exception('dmv termination failed')
+                logger.exception('dvm termination failed')
+                profiler.prof(event='orte_dvm_fail', uid=cfg['pilot_id'])
```
Comment (iparask) [inline src/radical/pilot/agent/lm/orte.py]: Do you think it is correct to have the same event name between a `ORTE DVM` failure and a shutdown failure?

File: src/radical/pilot/agent/lm/orte.py [hunk 2]
```diff
@@ -180,9 +184,12 @@ def lrms_shutdown_hook(cls, name, cfg, lrms, lm_info, logger):
                 orterun = ru.which('orterun')
                 if not orterun:
                     raise Exception("Couldn't find orterun")
-                subprocess.Popen([orterun, "--hnp", lm_info['dvm_uri'], "--terminate"])
+                ru.sh_callout('%s --hnp %s --terminate'
```
Comment (iparask) [inline src/radical/pilot/agent/lm/orte.py]: Is this relevant to this PR?

File: src/radical/pilot/agent/lm/fork.py [hunk 3]
```diff
@@ -26,7 +26,7 @@ def _configure(self):
     # --------------------------------------------------------------------------
     #
     @classmethod
-    def lrms_config_hook(cls, name, cfg, lrms, logger):
+    def lrms_config_hook(cls, name, cfg, lrms, logger, profile):
```
Comment (iparask) [inline src/radical/pilot/agent/lm/fork.py]: Here also it is `profile`. I think it is better if it is profiler

File: src/radical/pilot/agent/lm/orte_lib.py [hunk 4]
```diff
@@ -184,10 +187,13 @@ def lrms_shutdown_hook(cls, name, cfg, lrms, lm_info, logger):
                 orte_submit = ru.which('orterun')
                 if not orte_submit:
                     raise Exception("Couldn't find orterun")
-                subprocess.Popen([orte_submit, "--hnp", lm_info['dvm_uri'], "--terminate"])
+                ru.sh_callout('%s --hnp %s --terminate'
```
Comment (iparask) [inline src/radical/pilot/agent/lm/orte_lib.py]: Same as above.

File: src/radical/pilot/agent/lm/orte.py (hunk 1 above)
Comment (andre-merzky) [reply src/radical/pilot/agent/lm/orte.py]: Actually yes - since the DVM is running asynchronously, we can't really distinguish a shutdown failure from a runtime failure.  In fact, the most common case where we see the shutdown failure is when the shutdown complains that the DVM is already gone, which is then indeed a runtime failure.  So at the moment this is intentional for the profiler, but the logs will distinguish it.  I added a comment in 64cf2da5 to clarify this.

**Comment by andre-merzky** [comment]
The `profiler` / `profile` parameter mixup is a good catch - this is fixed in f31b60d7.  Thanks!

File: src/radical/pilot/agent/lm/fork.py (hunk 3 above)
Comment (andre-merzky) [reply src/radical/pilot/agent/lm/fork.py]: fixed in 64ba387d - thanks.

File: src/radical/pilot/agent/lm/orte.py (hunk 2 above)
Comment (andre-merzky) [reply src/radical/pilot/agent/lm/orte.py]: I hear you, good question, and I assume a matter of taste.  I tend to try to improve code pieces where there are obvious legacy things missed by previous refactoring and stuff as long as it is as small as here (single line).  If that affects more code and convolutes the PR, it should got into a separate PR indeed.

Happy to jump through the exercise to move it out.  To answer explicitly: no, that change is cosmetic, and not related to the PR otherwise.

File: src/radical/pilot/agent/lm/orte_lib.py (hunk 4 above)
Comment (andre-merzky) [reply src/radical/pilot/agent/lm/orte_lib.py]: same reply :)

**Comment by andre-merzky** [comment]
test this please

**Comment by andre-merzky** [comment]
jenkins test this please

**Comment by iparask** [comment]
You want jenkins to do or me?

**Comment by andre-merzky** [comment]
That was supposed to trigger Jenkins, not you :-)
